packages/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -393,7 +393,7 @@ const GridCell = React.forwardRef<HTMLDivElement, GridCellProps>(function GridCe
       <div
         data-colindex={colIndex}
         role="presentation"
-        style={{ ...style, minWidth: 'var(--width)', maxWidth: 'var(--width)' }}
+        style={{ width: 'var(--width)', ...style }}
```

Comment:
The spread should come before the `width`, I don't think we want to allow users to update the width.

Reply:
I have used this order to replicate the CSS override order with regular cells. Applied with https://github.com/oliviertassinari/mui-x/blob/79836d43449603af4435182bca421507c09bcc0a/packages/x-data-grid/src/components/containers/GridRootStyles.ts#L479 so lose to inline style.